network-api/networkapi/wagtailpages/factory/homepage.py

```diff
@@ -0,0 +1,94 @@
+from django.conf import settings
+from factory import (
+    Faker,
+    SubFactory
+)
+from wagtail_factories import (
+    PageFactory,
+    ImageFactory
+)
+from wagtail.core.models import (
+    Site as WagtailSite,
+    Page as WagtailPage
+)
+
+from networkapi.wagtailpages.models import Homepage
+from networkapi.utility.faker.helpers import (
+    reseed,
+    get_homepage
+)
+from .primary_page import PrimaryPageFactory
+
+if settings.HEROKU_APP_NAME:
+    REVIEW_APP_NAME = settings.HEROKU_APP_NAME
+    REVIEW_APP_HOSTNAME = f'{REVIEW_APP_NAME}.herokuapp.com'
+
+
+class WagtailHomepageFactory(PageFactory):
+    class Meta:
+        model = Homepage
+
+    hero_headline = Faker('text', max_nb_chars=140)
+    hero_story_description = Faker('paragraph', nb_sentences=5, variable_nb_sentences=True)
+    hero_button_text = Faker('text', max_nb_chars=50)
+    hero_button_url = Faker('url')
+    hero_image = SubFactory(ImageFactory)
+
+
+def generate(seed):
+    reseed(seed)
+
+    print('Generating blank Homepage')
+    try:
+        home_page = get_homepage()
+        print('Homepage already exists')
+    except Homepage.DoesNotExist:
+        print('Generating a Homepage')
+        site_root = WagtailPage.objects.get(title='Root')
```

Comment:
I get a crash here: delete db.sqlite3, `inv migrate`, `inv load_fake_data` yields no pages called `Root`, and so this line throws:
```
>inv manage load_fake_data
Loading .env environment variables…
Seeding random numbers with: 1410850
Generating Images
Generating Fake News
Generating Highlights
Generating five InternetHealthIssue
Generating People
Generating People with Affiliation
Generating blank Homepage
Generating a Homepage
wagtail site root could not be found.
title for page 1:
title for page 2:
Traceback (most recent call last):
  File "C:\Users\Mike\Documents\Git\collaborations\mozilla\foundation.mozilla.org\network-api\networkapi\wagtailpages\factory\homepage.py", line 44, in generate
    home_page = get_homepage()
  File "C:\Users\Mike\Documents\Git\collaborations\mozilla\foundation.mozilla.org\network-api\networkapi\utility\faker\helpers.py", line 56, in get_homepage
    raise ex
  File "C:\Users\Mike\Documents\Git\collaborations\mozilla\foundation.mozilla.org\network-api\networkapi\utility\faker\helpers.py", line 52, in get_homepage
    return Homepage.objects.get(title='Homepage')
  File "C:\Users\Mike\.virtualenvs\foundation.mozilla.org-4aDC7Y6j\lib\site-packages\django\db\models\manager.py", line 82, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "C:\Users\Mike\.virtualenvs\foundation.mozilla.org-4aDC7Y6j\lib\site-packages\django\db\models\query.py", line 408, in get
    self.model._meta.object_name
networkapi.wagtailpages.models.DoesNotExist: Homepage matching query does not exist.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "network-api/manage.py", line 22, in <module>
    execute_from_command_line(sys.argv)
  File "C:\Users\Mike\.virtualenvs\foundation.mozilla.org-4aDC7Y6j\lib\site-packages\django\core\management\__init__.py", line 381, in execute_from_command_line
    utility.execute()
  File "C:\Users\Mike\.virtualenvs\foundation.mozilla.org-4aDC7Y6j\lib\site-packages\django\core\management\__init__.py", line 375, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "C:\Users\Mike\.virtualenvs\foundation.mozilla.org-4aDC7Y6j\lib\site-packages\django\core\management\base.py", line 323, in run_from_argv
    self.execute(*args, **cmd_options)
  File "C:\Users\Mike\.virtualenvs\foundation.mozilla.org-4aDC7Y6j\lib\site-packages\django\core\management\base.py", line 364, in execute
    output = self.handle(*args, **options)
  File "C:\Users\Mike\Documents\Git\collaborations\mozilla\foundation.mozilla.org\network-api\networkapi\management\commands\load_fake_data.py", line
97, in handle
    buyersguide_factory
  File "C:\Users\Mike\Documents\Git\collaborations\mozilla\foundation.mozilla.org\network-api\networkapi\management\commands\load_fake_data.py", line
91, in <listcomp>
    [app_factory.generate(seed) for app_factory in [
  File "C:\Users\Mike\Documents\Git\collaborations\mozilla\foundation.mozilla.org\network-api\networkapi\wagtailpages\factory\__init__.py", line 16, in generate
    homepage.generate(seed)
  File "C:\Users\Mike\Documents\Git\collaborations\mozilla\foundation.mozilla.org\network-api\networkapi\wagtailpages\factory\homepage.py", line 48, in generate
    site_root = get_site_root()
  File "C:\Users\Mike\Documents\Git\collaborations\mozilla\foundation.mozilla.org\network-api\networkapi\utility\faker\helpers.py", line 46, in get_site_root
    raise ex
  File "C:\Users\Mike\Documents\Git\collaborations\mozilla\foundation.mozilla.org\network-api\networkapi\utility\faker\helpers.py", line 38, in get_site_root
    return WagtailPage.objects.get(title='Root')
  File "C:\Users\Mike\.virtualenvs\foundation.mozilla.org-4aDC7Y6j\lib\site-packages\django\db\models\manager.py", line 82, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "C:\Users\Mike\.virtualenvs\foundation.mozilla.org-4aDC7Y6j\lib\site-packages\django\db\models\query.py", line 408, in get
    self.model._meta.object_name
wagtail.core.models.DoesNotExist: Page matching query does not exist.
```
```

Reply:
This was my bad, I introduced a regression when I rebased the original load_fake_data refactor branch. We fixed this issue getting the root page before by searching for the page with `id=1` instead of `title='Root'` I suspect an older version of Wagtail might have given the root page a title at some point when I wrote the original load_fake_data command.